msticpy/data/sql_to_kql.py

```diff
@@ -0,0 +1,602 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+Module for SQL to KQL Conversion.
+
+This is an experiment conversion utility built to support a limited subset
+of ANSI SQL.
+It relies on moz_sql_parser (https://github.com/mozilla/moz-sql-parser)
+to parse the SQL syntax tree. Some hacky additions have been done to
+allow table renaming and support for a few SparkSQL operators such as
+RLIKE.
+
+For a more complete translation help with SQL to KQL see
+
+Known limitations
+-----------------
+
+- Does not support aggregate functions in SELECT with no GROUP BY clause
+- Does not support IN, EXISTS, HAVING operators
+- Only partial support for AS naming (should work in SELECT expressions)
+
+"""
+import random
+import re
+from typing import List, Tuple, Any, Union, Dict, Optional
+
+from ..common.exceptions import MsticpyImportExtraError
+
+try:
+    import moz_sql_parser
+    from moz_sql_parser import parse
+except ImportError as imp_err:
+    raise MsticpyImportExtraError(
+        "Cannot use this feature without moz_sql_parser installed",
+        title="Error importing moz_sql_parser for sql_to_kql",
+        extra="sql",
+    ) from imp_err
+
+
+from .._version import VERSION
+
+__version__ = VERSION
+__author__ = "Ian Hellen"
+
+_DEBUG = False
+
+SPARK_KQL_FUNC_MAP = {
+    "avg": ("avg", None, None),
+    "base64": ("base64_encode_tostring", None, None),
+    "concat": ("strcat", None, None),
+    "count": ("count", None, None),
+    "if": ("iif", None, None),
+    "iif": ("iif", None, None),
+    "ifnull": (None, None, "iif(isnull({p0}), {p1}, {p0}))"),
+    "in": ("in", None, None),
+    "instr": ("indexof", None, None),
+    "int": ("toint", None, None),
+    "isnotnull": ("isnotnull", None, None),
+    "isnull": ("isnull", None, None),
+    "left": ("NA", None, None),
+    "length": ("strlen", None, None),
+    "like": ("NA", None, None),
+    "locate": ("indexof", None, None),
+    "lower": ("tolower", None, None),
+    "lcase": ("tolower", None, None),
+    "ltrim": ("trim_start", None, None),
+    "max": ("max", None, None),
+    "mean": ("mean", None, None),
+    "min": ("min", None, None),
+    "position": ("indexof", None, None),
+    "regex_extract": ("extract", "{p1}, {p0}", None),  # swap params 0, 1
```

Comment:
I think its regexp_extract in Spark - https://spark.apache.org/docs/latest/api/sql/index.html#regexp_extract

Reply:
Good catch! Thanks